### ISSUE TITLE
Display errors defaults to true

### ISSUE THREAD

**Issue by maarten00** [pods-framework/pods#3060]
In the changelog for 2.5 it says:
* Changed the default of $display_errors in Pods class to true.

Is there a scenario where this is desired behavior? Why not just use WP_DEBUG to determine this? 
I for one certainly don't like that this causes SQL errors to be displayed to the end-user on my live server.

**Comment by pglewis** [comment]
This change was made in order to maintain backward compatibility with older code.  

The original code in general.php had the logic that checks `$display_errors` reversed: https://github.com/pods-framework/pods/commit/b71c31

This correction, however, changes the default behavior of existing code.  The only reason this is desired default behavior is to maintain the same behavior as before without requiring changes to existing code.

**Comment by pglewis** [comment]
I'm wondering if we should consider setting this back on 3.0.  The history is we originally got it wrong and now we're stuck with the behavior on the 2.x branch.  But I believe we might be able to justify this breakage of back-compat in 3.0.

**Comment by pglewis** [comment]
Related: https://github.com/pods-framework/pods/issues/3060

**Comment by maarten00** [comment]
I'm not sure I fully understand, Pods is meant to be used in production environments, right? I don't believe that any developer would want SQL errors to be shown to the end-user in a production environment. Logging errors is fine, displaying them should always be optional, the only possible exception would be your development environment. 

With that being said, is it possible to overwrite this behaviour in the meantime? I've searched excessively, but I can't seem to find anything on the subject .

**Comment by pglewis** [comment]
The original default for `$display_errors` in the `Pods` class was `false` and the intended behavior was as you describe.  But the logic for checking `$display_errors` in the `pods_error()` function was backwards, resulting in behavior as if `$display_errors` defaulted to `true`.

There is a fair codebase out there that was written around the incorrect default behavior so that when the logic in `pods_error()` was corrected we started breaking some existing code.  In order to avoid that we have reverted the default for `$display_errors` in the `Pods` class to `true`, a necessary evil.  

You should be able to explicitly set it to `false` but it probably depends on the context.  How are are you using it?

**Comment by maarten00** [comment]
We mainly use the the pods() function from classes.php with a bunch of arguments through an array. Setting a default with a define or something through the config would be preferred.

**Comment by pglewis** [comment]
I think a new constant is a good idea.  You could set that in your wp-config and forget about it, but it wouldn't break existing stuff.

**Comment by Ramoonus** [comment]
@pglewis what about using the wordpress debug constants?

**Comment by maarten00** [comment]
@pglewis I'm still looking for a solution, is there a way to default this to false globally in 2.5.4? We're using the pods() global function. I'd prefer not having to modify all of those calls, as we have over 60 websites that use Pods which all have this issue.

**Comment by pglewis** [comment]
I'd like to get something into 2.5.5 to workaround this unfortunate situation, assigning myself.

**Comment by pglewis** [comment]
I suggest something as simple as this:

```
if ( ! defined( 'PODS_GLOBAL_HIDE_ERRORS' ) || false === PODS_GLOBAL_HIDE_ERRORS ) {
	if ( is_object( $obj ) && isset( $obj->display_errors ) && true === $obj->display_errors ) {
		$display_errors = true;
	} elseif ( is_bool( $obj ) && true === $obj ) {
		$display_errors = true;
	}
}
```

https://github.com/pods-framework/pods/blob/72a4c6/includes/general.php#L117-L120

And for 3.0, I suggest we just break back compat for once and default to $display_errors = false in Pods the way God intended

**Comment by sc0ttkclark** [comment]
I think all of our errors are exceptions first. So if you try / catch, you'd catch the issue gracefully. I can get behind a more sane display_errors = false in 3.0

**Comment by pglewis** [comment]
Yeah, there are workarounds but the discussion here is use cases it would be super helpful to provide a global constant to turn it off site-wide in one place.
